### ISSUE TITLE
What is our popup story?

### ISSUE THREAD

**Issue by larsbergstrom** [servo/servo#10056]
@dati91 is currently working on step #2 in http://webbluetoothcg.github.io/web-bluetooth/#requestDevice-secure-context, which can require a popup.

For now, we can clearly just assume that popups are never allowed, but I noticed that we don't currently have any issues on:
1) Implementation of https://html.spec.whatwg.org/multipage/browsers.html#allowed-to-show-a-popup
2) What should servo _do_ for such popups?

CC @jdm @Ms2ger

**Comment by Ms2ger** [comment]
I don't understand why the bluetooth api looks at that.

As for popups themselves: can we open new tabs from within Servo yet? We'll need to figure out a design for that, but that's more constellation than script.

**Comment by paulrouget** [comment]
When that makes sense, we should delegate that to app itself, maybe via the browser api (when iframe are used):
- https://developer.mozilla.org/en-US/docs/Web/Events/mozbrowsershowmodalprompt
- https://developer.mozilla.org/en-US/docs/Web/Events/mozbrowserusernameandpasswordrequired
- https://developer.mozilla.org/en-US/docs/Web/Events/mozbrowseropentab
- https://developer.mozilla.org/en-US/docs/Web/Events/mozbrowseropenwindow

**Comment by Ms2ger** [comment]
As long as we don't end up with anything resembling the horror show that's Gecko window opening, I'm happy.

**Comment by notriddle** [comment]
Do we need to have some sort of sensible behavior if browser.html itself tries to show a modal popup?

**Comment by jdm** [comment]
Yeah, that would be the same as some random website performing the same action in non-browser.html.

**Comment by jdm** [comment]
In terms of Servo's current abilities, we have no support for multiple windows yet. IIRC [miniservo-gtk](https://github.com/glennw/miniservo-gtk/) supported multiple tabs by instantiating a new Constellation for each tab.

**Comment by paulrouget** [comment]
We gave up on browserhtml. The new approach is to let the embedder know that it needs to show a popup. See the recent `alert` implementation: #20707.

**Comment by lumiscosity** [comment]
> We gave up on browserhtml. The new approach is to let the embedder know that it needs to show a popup. See the recent `alert` implementation: [#20707](https://github.com/servo/servo/pull/20707).

I suppose this means this is done now?

<img width="691" height="556" alt="Image" src="https://github.com/user-attachments/assets/cb00674a-17db-420b-b4d6-1fdc6e5f6238" />

(example of servoshell using this API to display an alert with egui)

**Comment by jdm** [comment]
Yeah, we can file new issues about anything that doesn't work as expected. The links in the original issue description don't resolve any more.
